pkg/tests/tasks/traffic/egress/egress_gateways_tls_sds_test.go

```diff
@@ -15,110 +15,81 @@
-		util.KubeDeleteContents("bookinfo", util.RunTemplate(ExGatewayTLSFileTemplate, smcp))
-		util.KubeDeleteContents("bookinfo", ExServiceEntry)
-		time.Sleep(time.Duration(20) * time.Second)
-	})
-
-	t.Run("TrafficManagement_egress_gateway_perform_mtls_origination", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Deploy nginx mtls server")
-		nginx := examples.Nginx{Namespace: "mesh-external"}
-		nginx.Install_mTLS(env.GetRootDir() + "/testdata/examples/x86/nginx/nginx_mesh_external_ssl.conf")
-
-		log.Log.Info("Create client cert secret")
-		util.Shell(`kubectl create secret -n %s generic client-credential --from-file=tls.key=%s --from-file=tls.crt=%s --from-file=ca.crt=%s`,
-			meshNamespace,
-			nginxClientCertKey,
-			nginxClientCert,
-			nginxServerCACert)
-
-		log.Log.Info("Configure MTLS origination for egress traffic")
-		util.KubeApplyContents("bookinfo", util.RunTemplate(EgressGatewaySDSTemplate, smcp))
-		util.KubeApplyContents(meshNamespace, meshExternalServiceEntry)
-		util.KubeApplyContents(meshNamespace, OriginateSDS)
-		time.Sleep(time.Duration(10) * time.Second)
-
-		log.Log.Info("Verify NGINX server")
-		cmd := fmt.Sprintf(`curl -sS http://my-nginx.mesh-external.svc.cluster.local`)
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", cmd, true)
-		util.Inspect(err, "failed to get response", "", t)
-		if !strings.Contains(msg, "Welcome to nginx") {
-			t.Errorf("Expected Welcome to nginx; Got unexpected response: %s", msg)
-			log.Log.Errorf("Expected Welcome to nginx; Got unexpected response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get expected response: %s", msg)
-		}
+	NewTest(t).Id("T15").Groups(Full, InterOp).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		ns := "bookinfo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+		})
+
+		app.InstallAndWaitReady(t, app.Sleep(ns))
+
+		t.NewSubTest("ServiceEntry").Run(func(t TestHelper) {
+			t.LogStep("Perform TLS origination with an egress gateway")
+			oc.ApplyString(t, ns, ExServiceEntry)
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, ExServiceEntry)
+			})
+			execInSleepPod(t, ns,
+				`curl -sSL -o /dev/null -D - http://istio.io`,
+				assert.OutputContains(
+					"301",
+					"Expected 301 Moved Permanently",
+					"Unexpected response, expected 301 Moved Permanently"))
+
+			t.LogStep("Create a Gateway to external istio.io")
+			oc.ApplyTemplate(t, ns, ExGatewayTLSFileTemplate, smcp)
+			t.Cleanup(func() {
+				oc.DeleteFromTemplate(t, ns, ExGatewayTLSFileTemplate, smcp)
+			})
+			execInSleepPod(t, ns,
+				`curl -sSL -o /dev/null -D - http://istio.io`,
+				assert.OutputContains(
+					"HTTP/1.1 200 OK",
+					"Expected 200 from istio.io",
+					"Unexpected response, expected 200"))
+		})
+
+		t.NewSubTest("Gateway").Run(func(t TestHelper) {
+			t.Log("Perform mTLS origination with an egress gateway")
+			nsNginx := "mesh-external"
+			t.Cleanup(func() {
+				oc.DeleteNamespace(t, nsNginx)
```

Comment:
This must recreate the namespace, otherwise the next test in the same package might fail (because the setup code that creates these namespaces is only called once per package). This also prevents you from running this test multiple times. 
```suggestion
				oc.RecreateNamespace(t, nsNginx)
```